tool/tctl/common/scoped_token_command.go

```diff
@@ -0,0 +1,542 @@
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if len(proxies) == 0 {
+			return trace.NotFound("cluster has no proxies")
+		}
+		return dbMessageTemplate.Execute(c.Stdout,
+			map[string]any{
+				"token":        tokenName,
+				"scope":        tok.GetScope(),
+				"assign_scope": tok.GetSpec().GetAssignedScope(),
+				"minutes":      c.ttl.Minutes(),
+				"ca_pins":      caPins,
+				"auth_server":  proxies[0].GetPublicAddr(),
+				"db_name":      c.dbName,
+				"db_protocol":  c.dbProtocol,
+				"db_uri":       c.dbURI,
+			})
+	case roles.Include(types.RoleTrustedCluster):
+		fmt.Fprintf(c.Stdout, trustedClusterMessage,
+			tokenName,
+			int(c.ttl.Minutes()))
+	case roles.Include(types.RoleWindowsDesktop):
+		return desktopMessageTemplate.Execute(c.Stdout,
+			map[string]any{
+				"token":        tokenName,
+				"scope":        tok.GetScope(),
+				"assign_scope": tok.GetSpec().GetAssignedScope(),
+				"minutes":      c.ttl.Minutes(),
+			})
+	case roles.Include(types.RoleMDM):
+		return mdmTokenAddTemplate.Execute(c.Stdout, map[string]any{
+			"token":        tokenName,
+			"scope":        tok.GetScope(),
+			"assign_scope": tok.GetSpec().GetAssignedScope(),
+			"minutes":      c.ttl.Minutes(),
+			"ca_pins":      caPins,
+		})
+	default:
+		authServer := authServers[0].GetAddr()
+
+		pingResponse, err := client.Ping(ctx)
+		if err != nil {
+			slog.DebugContext(ctx, "unable to ping auth client", "error", err)
+		}
+
+		if err == nil && pingResponse.GetServerFeatures().Cloud {
+			proxies, err := client.GetProxies()
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			if len(proxies) != 0 {
+				authServer = proxies[0].GetPublicAddr()
+			}
+		}
+
+		return nodeMessageTemplate.Execute(c.Stdout, map[string]any{
+			"token":        tokenName,
+			"scope":        tok.GetScope(),
+			"assign_scope": tok.GetSpec().GetAssignedScope(),
+			"roles":        strings.ToLower(roles.String()),
+			"minutes":      int(c.ttl.Minutes()),
+			"ca_pins":      caPins,
+			"auth_server":  authServer,
+		})
+	}
+
+	return nil
+}
+
+// Del is called to execute "scoped tokens del ..." command.
+func (c *ScopedTokensCommand) Del(ctx context.Context, client *authclient.Client) error {
+	if c.value == "" {
+		return trace.BadParameter("Need an argument: token")
+	}
+	if err := client.DeleteScopedToken(ctx, c.value); err != nil {
+		return trace.Wrap(err)
+	}
+	fmt.Fprintf(c.Stdout, "Token %s has been deleted\n", c.value)
+	return nil
+}
+
+func (c *ScopedTokensCommand) getFilter(filter string) (*scopesv1.Filter, error) {
+	modeOrScope, scope, found := strings.Cut(filter, ":")
+
+	if !found {
+		return &scopesv1.Filter{
+			Mode:  scopesv1.Mode_MODE_RESOURCES_SUBJECT_TO_SCOPE,
+			Scope: modeOrScope,
+		}, nil
+	}
+	var mode scopesv1.Mode
+	switch strings.ToLower(modeOrScope) {
+	case "ancestor":
+		mode = scopesv1.Mode_MODE_POLICIES_APPLICABLE_TO_SCOPE
+	case "descendant":
+		mode = scopesv1.Mode_MODE_RESOURCES_SUBJECT_TO_SCOPE
```

Comment:
I don't think we've settled on how to present these concepts to the user, but `ancestor`/`descendant` terminology isn't quite accurate.  I'm actually uncertain if we will ever need to present these options directly to the user.  I'd suggest omitting filter support in the list command for now.